### PR TITLE
PR: Fix circle and appveyor tests

### LINKED ISSUE #602: Increase minimal supported Python 3 version to 3.5
Python 3.4 reached end of life last month, so there's no need to support it anymore. We also need to move to 3.5 in Appveyor and CircleCI.

@steff456, please take care of this one.

### PR DESCRIPTION
**Pull request by steff456**
Fixes #602

### CODE REVIEW

**Comment by palantirtech** [comment]
Thanks for your interest in [palantir/python-language-server](https://github.com/palantir/python-language-server), @steff456! Before we can accept your pull request, you need to sign our contributor license agreement - just visit https://cla.palantir.com/ and follow the instructions. Once you sign, I'll automatically update this pull request.

File: .circleci/config.yml [hunk 1]
```diff
@@ -14,7 +14,7 @@ jobs:
 
   python3-test:
     docker:
-      - image: "python:3.4-stretch"
+      - image: "python:3.7-stretch"
```
Comment (ccordoba12) [inline .circleci/config.yml]: Let's go with Python 3.5 instead of Python 3.7

File: .circleci/config.yml [hunk 2]
```diff
@@ -32,7 +32,7 @@ jobs:
 
   publish:
     docker:
-      - image: "python:3.4-stretch"
+      - image: "python:3.7-stretch"
```
Comment (ccordoba12) [inline .circleci/config.yml]: 3.5 here too

File: appveyor.yml [hunk 3]
```diff
@@ -4,8 +4,8 @@ environment:
       PYTHON_VERSION: "2.7.15"
       PYTHON_ARCH: "64"
 
-    - PYTHON: "C:\\Python34"
-      PYTHON_VERSION: "3.4.9"
+    - PYTHON: "C:\\Python37"
+      PYTHON_VERSION: "3.7.3"
```
Comment (ccordoba12) [inline appveyor.yml]: And also here 3.5

**Review by gatesn — APPROVED** [review]

**Pull request merged**